lib/pacemaker/pcmk_sched_utilization.c

```diff
@@ -195,43 +195,67 @@ pcmk__release_node_capacity(GHashTable *current_utilization, pe_resource_t *rsc)
 }
 
 
+/*
+ * Functions for checking for sufficient node capacity
+ */
+
 struct capacity_data {
     pe_node_t *node;
     const char *rsc_id;
-    gboolean is_enough;
+    bool is_enough;
 };
 
+/*!
+ * \internal
+ * \brief Check whether a single utilization attribute has sufficient capacity
+ *
+ * \param[in] key        Name of utilization attribute to check
+ * \param[in] value      Amount of utilization required
+ * \param[in] user_data  Capacity data (as struct capacity_data *)
+ */
 static void
 check_capacity(gpointer key, gpointer value, gpointer user_data)
 {
     int required = 0;
     int remaining = 0;
+    const char *node_value_s = NULL;
     struct capacity_data *data = user_data;
 
+    node_value_s = g_hash_table_lookup(data->node->details->utilization, key);
+
     required = utilization_value(value);
-    remaining = utilization_value(g_hash_table_lookup(data->node->details->utilization, key));
+    remaining = utilization_value(node_value_s);
 
     if (required > remaining) {
-        CRM_ASSERT(data->rsc_id);
-        CRM_ASSERT(data->node);
-
-        crm_debug("Node %s does not have enough %s for %s: required=%d remaining=%d",
-                  data->node->details->uname, (char *)key, data->rsc_id, required, remaining);
-        data->is_enough = FALSE;
+        crm_debug("Node %s remaining capacity (%d) for %s is insufficient "
+                  "for resource %s usage (%d)",
```

Comment:
How about "Remaining capacity (%d) on node %s for %s..."

Reply:
updated commit